adb_graphics/figure_builders.py

```diff
@@ -0,0 +1,280 @@
+            tile=tile,
+            )
+
+        # Send all objects (map_field, contours, hatches) to a DataMap object
+        dm = map_class(
+            map_fields=map_fields,
+            map_=m,
+            member=mem,
+            model_name=cla.model_name,
+            last_panel=last_panel
+            )
+
+        # Draw the map
+        if cla.graphic_type == 'enspanel':
+            if index == 0:
+                dm.title()
+                dm.add_logo(current_ax)
+            elif index == 8:
+                if spec.get('include_obs', False):
+                    # Add observation panel to lower left. Currently only
+                    # supported for composite reflectivity.
+                    add_obs_panel(
+                        ax=axes[8],
+                        model_name=cla.model_name,
+                        obs_file=cla.obs_file_path,
+                        proj_info=field.grid_info(),
+                        short_name=variable,
+                        tile=tile,
+                        )
+            else:
+                dm.draw(show=True)
+        else:
+            dm.draw(show=True)
+
+    # Build the output path
+    png_file = f'{variable}_{tile}_{level}_f{fhr:03d}.png'
+    png_file = png_file.replace("__", "_")
+    png_path = os.path.join(workdir, png_file)
+
+    print('*' * 120)
+    print(f"Creating image file: {png_path}")
+    print('*' * 120)
+
+    # Save the png file to disk
+    plt.savefig(
+        png_path,
+        bbox_inches='tight',
+        dpi=cla.img_res,
+        format='png',
+        orientation='landscape',
+        pil_kwargs={'optimize': True},
+        )
+
+    fig.clear()
+    # Clear the current axes.
+    plt.cla()
+    # Clear the current figure.
+    plt.clf()
+    # Closes all the figure windows.
+    plt.close('all')
+    del field
+    del m
+    gc.collect()
+
+def parallel_skewt(cla, fhr, ds, site, workdir):
+
+    '''
+    Function that creates a single SkewT plot. Can be used in parallel.
+    Input:
+
+      cla        command line arguments Namespace object
+      ds         the XArray dataset
+      fhr        the forecast hour integer
+      site       the string representation of the site from the sites file
+      workdir    output directory
+    '''
+
+    skew = skewt.SkewTDiagram(
+        ds=ds,
+        fhr=fhr,
+        filetype=cla.file_type,
+        loc=site,
+        max_plev=cla.max_plev,
+        model_name=cla.model_name,
+        )
+    skew.create_diagram()
+    outfile = f"{skew.site_code}_{skew.site_num}_skewt_f{fhr:03d}.png"
+    png_path = os.path.join(workdir, outfile)
+
+    print('*' * 80)
+    print(f"Creating image file: {png_path}")
+    print('*' * 80)
+
+    # pylint: disable=duplicate-code
+    plt.savefig(
+        png_path,
+        bbox_inches='tight',
+        dpi=cla.img_res,
+        format='png',
+        orientation='landscape',
+        )
+    plt.close()
+
+def set_figure(model_name, graphic_type, tile):
```

Comment:
This is a new one that sets the size and aspect ratio for the plots. Square doesn't work well for the panel plots, and the same rectangle isn't great for all the subdomains.